app/models/user.rb

```diff
@@ -14,10 +16,22 @@ class User < ApplicationRecord
 
   def initialize(attributes = {})
     super(attributes)
-    return unless person.nil?
+    if person.nil?
+      self.person = Person.new
+      person.email = email
+    end
+    person.owners = [self]
+  end
 
-    self.person = Person.new
-    person.email = email
+  def roles=(roles)
+    roles = [*roles].map(&:to_sym)
+    self.roles_mask = (roles & ROLES).sum { |r| 2**ROLES.index(r) }
+  end
+
+  def roles
+    ROLES.reject do |r|
+      ((roles_mask.to_i || 0) & (2**ROLES.index(r))).zero?
+    end
```

Comment:
not yet necessary because there aren't multiple roles right now, right?
Probably copies from here: https://github.com/CanCanCommunity/cancancan/blob/develop/docs/role_based_authorization.md

Reply:
Any chance of adding some explanatory comments for this?
`((roles_mask.to_i || 0) & (2**ROLES.index(r))).zero?` might not be immediately obvious

Reply:
We are too waiting for the PR owners :)

Reply:
> not yet necessary because there aren't multiple roles right now, right? Probably copied from here: https://github.com/CanCanCommunity/cancancan/blob/develop/docs/role_based_authorization.md#many-roles-per-user

Yes, it's from there and it's not completed. I think we can for now delete this, unless PO has some other Roles to apply this.

Reply:
We decided to remove this for now. Should be added in a separate PR when multiple roles are required.